config/application.rb

```diff
@@ -200,13 +200,6 @@ def try_config_for(*args)
     three_scale = config_for(:settings).symbolize_keys
     three_scale[:error_reporting_stages] = three_scale[:error_reporting_stages].to_s.split(/\W+/)
 
-    payment_settings = three_scale.extract!(:active_merchant_mode, :active_merchant_logging, :billing_canaries)
-    config.three_scale.payments = ActiveSupport::OrderedOptions.new
```

Comment:
In the spirit of retaining configuration as close to how it is done in clean rails, will our requirements be met if we introduce  e.g. `ThreeScale::OrderedOptions` that does our custom logic instead of a new global configuration singleton?